client/state/login/actions.js

```diff
@@ -113,26 +118,37 @@ export const loginUser = ( usernameOrEmail, password, rememberMe ) => dispatch =
 		} );
 };
 
+const getNonceTypeFromCode = ( code ) => {
+	switch ( code.length ) {
+		case 6:
+			return 'authenticator';
+		case 7:
+			return 'sms';
+		case 8:
+			return 'backup';
+	}
+};
+
 /**
  * Attempt to login a user when a two factor verification code is sent.
  *
- * @param  {Number}    user_id        Id of the user trying to log in.
  * @param  {String}    two_step_code  Verification code for the user.
- * @param  {String}    two_step_nonce Nonce generated for verification code submission.
  * @param  {Boolean}   remember_me       Flag for remembering the user for a while after logging in.
  * @return {Function}                 Action thunk to trigger the login process.
  */
-export const loginUserWithTwoFactorVerificationCode = ( user_id, two_step_code, two_step_nonce, remember_me ) => dispatch => {
+export const loginUserWithTwoFactorVerificationCode = ( two_step_code, remember_me ) => ( dispatch, getState ) => {
 	dispatch( { type: TWO_FACTOR_AUTHENTICATION_LOGIN_REQUEST } );
 
+	const nonceType = getNonceTypeFromCode( two_step_code );
```

Comment:
It seems we use the same convention on the server as well 🤔 . Nevermind then if you feel confident about it.